service/certconfig/v2/resources/vaultcrt/update.go

```diff
@@ -120,6 +122,18 @@ func (r *Resource) shouldCertBeRenewed(currentSecret, desiredSecret *apiv1.Secre
 		}
 	}
 
+	// Check if the cert configs ask to disable regeneration.
+	{
+		// TODO remove this hack once all cert configs are updated with the correct
+		// value for DisableRegeneration.
+		if customObject.Spec.Cert.ClusterComponent == string(certs.ServiceAccountCert) {
+			return false, nil
+		}
```

Comment:
This fixes our problem NOW. The condition below is for general purpose once all cert configs are up to date. This can still take a bit of time.